arch/nios2/include/kernel_arch_data.h

```diff
@@ -53,7 +53,7 @@ struct _kernel_arch {
 
 typedef struct _kernel_arch _kernel_arch_t;
 
-extern char _interrupt_stack[CONFIG_ISR_STACK_SIZE];
+extern K_THREAD_STACK_DEFINE(_interrupt_stack, CONFIG_ISR_STACK_SIZE);
```

Comment:
hmm I need to think about this some more.
DECLARE and DEFINE are really similar, at least with the 'extern' prefix it makes it really clear what's going on.